core/contextmenu_items.js

```diff
@@ -0,0 +1,272 @@
+/**
+ * @license
+ * Copyright 2020 Google LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @fileoverview Registers default context menu items.
+ * @author maribethb@google.com (Maribeth Bottorff)
+ */
+'use strict';
+
+/**
+ * @name Blockly.ContextMenuItems
+ * @namespace
+ */
+goog.provide('Blockly.ContextMenuItems');
+
+goog.requireType('Blockly.BlockSvg');
+
+/** Option to undo previous action. */
+Blockly.ContextMenuItems.registerUndo = function() {
+  var undoOption = {};
+  undoOption.displayText = function() {
+    return Blockly.Msg['UNDO'];
+  };
+  undoOption.preconditionFn = function(scope) {
+    if (scope.workspace.undoStack_.length > 0) {
+      return 'enabled';
+    }
+    return 'disabled';
+  };
+  undoOption.callback = function(scope) {
+    scope.workspace.undo(false);
+  };
+  undoOption.scopeType = Blockly.ContextMenuRegistry.ScopeType.WORKSPACE;
+  undoOption.id = 'undoWorkspace';
+  undoOption.weight = 0;
+  Blockly.ContextMenuRegistry.registry.register(undoOption);
+};
+
+/** Option to redo previous action. */
+Blockly.ContextMenuItems.registerRedo = function() {
+  var redoOption = {};
+  redoOption.displayText = function() { return Blockly.Msg['REDO']; };
+  redoOption.preconditionFn = function(scope) {
+    if (scope.workspace.redoStack_.length > 0) {
+      return 'enabled';
+    }
+    return 'disabled';
+  };
+  redoOption.callback = function(scope) {
+    scope.workspace.undo(true);
+  };
+  redoOption.scopeType = Blockly.ContextMenuRegistry.ScopeType.WORKSPACE;
+  redoOption.id = 'redoWorkspace';
+  redoOption.weight = 0;
+  Blockly.ContextMenuRegistry.registry.register(redoOption);
+};
+    
+/** Option to clean up blocks. */
+Blockly.ContextMenuItems.registerCleanup = function() {
+  var cleanOption = {};
+  cleanOption.displayText = function() {
+    return Blockly.Msg['CLEAN_UP'];
+  };
+  cleanOption.preconditionFn = function(scope) {
+    if (scope.workspace.isMovable()) {
+      if (scope.workspace.getTopBlocks(false).length > 1) {
+        return 'enabled';
+      }
+      return 'disabled';
+    }
+    return 'hidden';
+  };
+  cleanOption.callback = function(scope) {
+    scope.workspace.cleanUp();
+  };
+  cleanOption.scopeType = Blockly.ContextMenuRegistry.ScopeType.WORKSPACE;
+  cleanOption.id = 'cleanWorkspace';
+  cleanOption.weight = 0;
+  Blockly.ContextMenuRegistry.registry.register(cleanOption);
+};
+  
+/**
+ * Creates a callback to collapse or expand top blocks.
+ * @param {boolean} shouldCollapse Whether a block should collapse.
+ * @param {!Array<Blockly.BlockSvg>} topBlocks Top blocks in the workspace.
+ */
+var toggleOption = function(shouldCollapse, topBlocks) {
```

Comment:
Move these helper functions into the `Blockly.ContextMenuItems` namespace so that you don't populate the global namespace.

i.e. `toggleOption` should be `Blockly.ContextMenuItems.toggleOption`.

Also mark them private.